src/ILLink.RoslynAnalyzer/DiagnosticDescriptors.cs

```diff
@@ -68,6 +68,9 @@ static string GetDiagnosticCategory (DiagnosticId diagnosticId)
 			case >= 3050 and <= 6000:
 				return DiagnosticCategory.AOT;
 
+			case 1013:
+				return DiagnosticCategory.XML;
```

Comment:
I think the CategoryRanges were made to englobe similar diagnostics, 0 to 1000 is invalid, 1000 to 2000 is trimming errors, 2000 to 3000 is trimming warnings, etc. Xml is inherently part of the trimming, so maybe we could have a subcategory instead? (see https://github.com/dotnet/linker/blob/main/src/ILLink.Shared/DiagnosticId.cs#L194-L210 and https://github.com/dotnet/linker/blob/main/src/ILLink.Shared/MessageSubCategory.cs)

I guess the big difference would be on the suppression side, our code fixers use the diagnostic category to create the UnconditionalSuppressMessage attribute values.